### PR TITLE
App views PR

### PR DESCRIPTION
**Pull request by Javed-Shahbaz**
## In this project:
- Add the Views associated with the Models and Controllers for Blog App.
- Use preprocessed HTML files with embedded Ruby code.
- Use layouts and templates for shared content.

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1 +1,17 @@
...
+    <div class="users-container" >
+      <% @users.each do |user| %>
+    
+      <div class="single-user">
+        <div class="user-image">
+          <img src="<%= user.photo%>" alt="profile" class="user-profile">
+        </div>
+        <div class="user-information">
+          <h2 class="user-name"><%= user.name%></h2>
+            <span class="post-counter"><p class="count">Number of posts: <%= user.posts.count %></p></span>
```
Comment (tugsem) [inline app/views/users/index.html.erb]: - [ ] As per the requirement, clicking on each user must redirect us to the user show page. Kindly implement this.

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1 +1,35 @@
...
+            </div>
+        </div>
+        <div class="user-bio">
+            <h4 class="bio">Bio</h4>
+            <p class="bio-paragraph">
+                <%= @user.bio %>
+            </p>
+        </div>
+        <div class="user-all-posts">
+            <% @user.posts.each_with_index do |post,index|%>
```
Comment (tugsem) [inline app/views/users/show.html.erb]: - [ ] As per the requirement, only the 3 most recent posts of the user must be rendered here. Kindly implement this.
image
(your model)

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -1 +1,35 @@
...
+                    <p class="post-paragraph">
+                        <%= post.text %>
+                    </p>
+                    <span class="posts-count"><p class="count">Comments: <%=post.comments_counter%>, Likes: <%=post.likes_counter%></p></span>
+                </div>
+            </div>
+            <%end%>
+        </div>
+        <div class="see-all">
+            <button class="all">See all posts</button>
```
Comment (tugsem) [inline app/views/users/show.html.erb]: - [ ] As per the requirement, this button must redirect to the user post index page. Kindly add this.

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -1 +1,31 @@
...
+                        <p class="post-paragraph">
+                            <%= post.text %>
+                        </p>
+                        <span class="posts-count"><p class="count">Comments: <%= post.comments_counter %>, Likes: <%= post.likes_counter %></p></span>
+                    </div>
+                </div>
+                <div class="posts-comment">
+                    <% post.comments.each do |comment| %>
+                        <p class="comments"><%= comment.author.name %> : <%= comment.text %></p>
+                    <% end %>
```
Comment (tugsem) [inline app/views/posts/index.html.erb]: - [ ] As per the requirement, only the 5 most recent comments for the post must be rendered here. Kindly implement this.
image
(your model)

File: app/views/posts/index.html.erb [hunk 5]
```diff
@@ -1 +1,31 @@
...
+                <span class="post-counter"><p class="count">Number of posts: <%= @user.posts.count %></p></span>
+            </div>
+        </div>
+        <div class="user-all-posts">
+            <% @posts.each_with_index do |post, index| %>
+                <div class="user-post">
+                    <h4 class="post">Post #<%= index+1 %></h4>
+                    <div class="post-content">
+                        <p class="post-paragraph">
+                            <%= post.text %>
```
Comment (tugsem) [inline app/views/posts/index.html.erb]: - [ ] Please add a link to each post that redirects users to the post show page, which you will render all of the comments for the post.

**Review by tugsem — CHANGES_REQUESTED** [review]
##  STATUS: REQUIRED CHANGES ♻️
Hello 👋 , 

Good job so far! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text
## Highlights 🚀 
✔️controller and views for post and user created correctly
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20230728031306_create_users.rb
Comment (joskalenda) [inline db/migrate/20230728031306_create_users.rb:1]: - [ ] I checked your migration files and indeed there are duplicated files for creating user, posts, and comments migration file, this cause an error on running the migration and testing your app, please make sure you keep a unique migration file for each table and resubmit your work


CreateUsers
[1](https://github.com/Javed-Shahbaz/Blog-App/pull/6/commits/ca6be6f4c6e9b37e1e50db0b0b28ae04de01cb84#diff-d6a3b454d819869879b93cd70148c678e92bd28417ff3ca1abb4e100aea3f020) and [2](https://github.com/Javed-Shahbaz/Blog-App/blob/App-Views/db/migrate/20230728152703_create_users.rb)

CreatePosts
[1]() and [2](https://github.com/Javed-Shahbaz/Blog-App/blob/App-Views/db/migrate/20230728031343_create_posts.rb)

CreateComments
[1](https://github.com/Javed-Shahbaz/Blog-App/blob/App-Views/db/migrate/20230728032028_create_comments.rb) and [2](https://github.com/Javed-Shahbaz/Blog-App/blob/App-Views/db/migrate/20230728160517_create_comments.rb)

**Review by joskalenda — CHANGES_REQUESTED** [review]
| STATUS: REQUIRED CHANGES ♻️  |
|--------------------------------|
| Hi @Javed-Shahbaz , |
|Great work on making the changes requested by a previous reviewer 👏, You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet. |
|  |

## Highlights

- [ ]... ✅ Clean code
- [ ].. ✅ Green linters
- [ ].... ✅ Read me

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

-------
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me(@joskalenda) in your question so I can receive the notification.

> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by IngDavidTM — APPROVED** [review]
Hi @Javed-Shahbaz 🙋‍♂️, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request follows the GitFlow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**